src/main/java/net/sf/jabref/gui/BasePanel.java

```diff
@@ -2405,12 +2406,20 @@ public BibDatabaseContext getDatabaseContext() {
         return bibDatabaseContext;
     }
 
-    public SearchQuery getCurrentSearchQuery() {
+    public Optional<SearchQuery> getCurrentSearchQuery() {
         return currentSearchQuery;
     }
 
+    /**
+     * Set the query the user currently searches while this basepanel is active
+     * @param currentSearchQuery can be null
+     */
     public void setCurrentSearchQuery(SearchQuery currentSearchQuery) {
-        this.currentSearchQuery = currentSearchQuery;
+        if (currentSearchQuery == null) {
```

Comment:
However, I am unsure whether `trim().isEmpty()` should be checked for. I think, your current code is OK. (After the usage of `ofNullable`.